frontstage/views/surveys.py

```diff
@@ -247,7 +247,6 @@ def upload_survey(session):
         logger.debug('"event" : "Upload successful"')
         return render_template('surveys/surveys-upload-success.html', _theme='default', upload_filename=upload_filename)
     else:
-        logger.debug('"event" : "Upload failed"')
-        error_info = json.loads(result.text)
-        return render_template('surveys/surveys-upload-failure.html',  _theme='default', error_info=error_info,
-                               case_id=case_id)
+        logger.error('Upload failed. url: {}, status code: {}'.format(url, result.status_code))
+        error_info = {'status code': result.status_code, 'text': result.text}
+        return render_template('surveys/surveys-upload-failure.html', _theme='default', error_info=error_info, case_id=case_id)
```

Comment:
According to coverage this code isn't tested but that could be done in a separate PR probably as it wasn't done before/is subject to change